comp/core/workloadmeta/filter.go

```diff
@@ -91,7 +93,7 @@ func (fb *FilterBuilder) SetEventType(eventType EventType) *FilterBuilder {
 // MatchEntity returns true if the filter matches the passed entity.
 // If the filter is nil, or has no kinds, it always matches.
 func (f *Filter) MatchEntity(entity *Entity) bool {
-	if f == nil || len(f.kinds) == 0 {
```

Comment:
Shouldn't we keep the nil check? It's in other functions below.
I know that before we were using some nil filters. Not sure if after the refactor it's still the case, but in the feature someone could create a nil filter.

Reply:
Yes sorry it was dropped by mistake, fixing it

Reply:
`f.Kinds()` already handle the case of nil filter. 

Do you think it is necessary to duplicate the nil check here?

Reply:
I thought we were still checking len(f.kinds). All good :+1:

Reply:
Actually it's needed. Otherwise, `f.Kinds()` will panic if f is nil.

Reply:
The first line of `Kinds()` method is:
`if f == nil || len(f.kinds) == 0` 

I think this won't cause a panic. It will directly see that `f==nil` is true and return an empty list.